sdks/java/io/solr/src/test/java/org/apache/beam/sdk/io/solr/BeamThreadsFilter.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.io.solr;
+
+import com.carrotsearch.randomizedtesting.ThreadFilter;
+import com.google.common.annotations.VisibleForTesting;
+
+/**
+ * Filter out Beam leak threads.
```

Comment:
Solr-test-framework after running tests will try to detect leak threads. It seems that this line create a thread with name is `direct-metrics-counter-committer`
```
PAssert.thatSingleton(output.apply("Count", Count.<SolrDocument>globally()))
```
The Solr-test-framework after waiting for some time, do not see this thread get terminated. I do not sure this is a bug in Beam or not. But I did see same problem with testing solr in HDFS.